lib/upgrade_lts_contract.py

```diff
@@ -116,6 +117,11 @@ def process_contract_delta_after_apt_lock() -> None:
     logging.debug(msg)
 
 
+def remove_private_esm_apt_cache():
+    remove_folder(ESM_APT_ROOTDIR)
```

Comment:
hm, yes. I would say I agree to all of this.
However, I wonder what is actionable about it.

- importing the modules instead of specific functions make all sense and I will change it.
- the thing with forgiveness instead of permission is not, or at least was not, respected all over the codebase (I have myself seen reviews on PRs here say the opposite, suggesting putting guards instead of catching exceptions). Apart from being correct or agreeable, it is what it is. This can be "fixed" in a subsequent PR, and of course enforced through other PR reviews anyway.
- Well well this is the old discussion about logging EVERYTHING - it is a team preference and again, apart from being correct or if I agree or not, it is what it is. Good news here is that on this cycle we are taking the first steps to rework the logging - so re-evaluating this is a matter of time.
- And, about the anti-pattern of masking standard python functionality in utility functions... exactly the same. I personally disagree with this kind of implementation as well but again, this is how the team works and I would like to hear and discuss with everyone before changing the implemented code.
 
For this PR specifically, I will change the method name to be more expressive about removing the folder, as commented below with the `remove_file`. But I will keep the implementation there, as there is no point changing it here without also changing the implementation of such methods through the rest of the codebase.